code/build/template/include/extensions/[+python orxPy.h]

```diff
@@ -0,0 +1,2613 @@
+    BIND_FUNC("set_uint(key: str, value: int) -> None", set_uint);
+    BIND_FUNC("get_uint(key: str, index: int | None = None) -> int", get_uint);
+    BIND_FUNC("set_float(key: str, value: float) -> None", set_float);
+    BIND_FUNC("get_float(key: str, index: int | None = None) -> float", get_float);
+    BIND_FUNC("set_string(key: str, value: str) -> None", set_string);
+    BIND_FUNC("get_string(key: str, index: int | None = None) -> str", get_string);
+    BIND_FUNC("set_vector(key: str, value: Vector) -> None", set_vector);
+    BIND_FUNC("get_vector(key: str, index: int | None = None) -> Vector", get_vector);
+
+    BIND_FUNC("has_section(name: str) -> bool", has_section);
+    BIND_FUNC("has_value(key: str, check_spelling: bool = True) -> bool", has_value);
+
+    BIND_FUNC("clear_section(name: str) -> None", clear_section);
+    BIND_FUNC("clear_value(key: str) -> None", clear_value);
+
+#undef BIND_FUNC
+  }
+
+  void orxPy_AddModuleCommand()
+  {
+    // Register command module
+    py_Ref pyMod = py_newmodule("orx.command");
+
+#define BIND_FUNC(sig, name) py_bind(pyMod, sig, BINDING_NAME(Command, name))
+
+    // Bind command functions
+    BIND_FUNC("evaluate(command: str, guid: int | None = None) -> None", evaluate);
+
+#undef BIND_FUNC
+  }
+
+  void orxPy_AddModuleInput()
+  {
+    // Register command module
+    py_Ref pyMod = py_newmodule("orx.input");
+
+#define BIND_FUNC(sig, name) py_bind(pyMod, sig, BINDING_NAME(Input, name))
+
+    py_Type pyContextType = py_newtype("Set", tp_object, pyMod, orxNULL);
+
+    py_bindmagic(pyContextType, __new__, BINDING_NAME(Input, context__new__));
+    py_bindmagic(pyContextType, __enter__, BINDING_NAME(Input, context__enter__));
+    py_bindmagic(pyContextType, __exit__, BINDING_NAME(Input, context__exit__));
+
+    // Bind input functions
+    BIND_FUNC("push_set(name: str) -> None", push_set);
+    BIND_FUNC("pop_set() -> None", pop_set);
+    BIND_FUNC("enable_set(name: str, enable: bool = True) -> None", enable_set);
+    BIND_FUNC("is_set_enabled(name: str) -> bool", is_set_enabled);
+
+    BIND_FUNC("get_all() -> list[str]", get_all);
+
+    BIND_FUNC("is_active(name: str) -> bool", is_active);
+    BIND_FUNC("has_been_activated(name: str) -> bool", has_been_activated);
+    BIND_FUNC("has_been_deactivated(name: str) -> bool", has_been_deactivated);
+
+    BIND_FUNC("get_value(name: str) -> float", get_value);
+    BIND_FUNC("set_value(name: str, value: float, permanent: bool = False) -> None", set_value);
+    BIND_FUNC("reset_value(name: str) -> None", reset_value);
+
+#undef BIND_FUNC
+  }
+
+#undef BINDING_NAME
+
+  void orxPy_AddModules()
+  {
+    orxPy_AddModuleOrx();
+    orxPy_AddModuleClock();
+    orxPy_AddModuleVector();
+    orxPy_AddModuleConfig();
+    orxPy_AddModuleCommand();
+    orxPy_AddModuleInput();
+    orxPy_AddModuleObject();
+  }
+
+  orxSTATUS orxPy_EventHandler(const orxEVENT *_pstEvent)

```

Comment:
This is the event handler for all events with potential Python handlers.

The `if (sstPy.py* != 0)` checks will evaluate to a false value if no handler is defined in Python.